libs/cardano-data/src/Data/MonoTuple.hs

```diff
@@ -0,0 +1,48 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+
+-- | Supports creating tuples (where all components have the same type) as Fixed length lists.
+module Data.MonoTuple (
+  TupleN,
+  MonoTuple (list),
+) where
+
+import GHC.TypeNats (KnownNat, Nat)
+
+-- =========================================
+
+type family TupleN (n :: Nat) a where
+  TupleN 0 a = ()
+  TupleN 1 a = a
+  TupleN 2 a = (a, a)
+  TupleN 3 a = (a, a, a)
+  TupleN 4 a = (a, a, a, a)
+  TupleN 5 a = (a, a, a, a, a)
+  TupleN 6 a = (a, a, a, a, a, a)
+  TupleN 7 a = (a, a, a, a, a, a, a)
+  TupleN 8 a = (a, a, a, a, a, a, a, a)
+  TupleN 9 a = (a, a, a, a, a, a, a, a, a)
+  TupleN 10 a = (a, a, a, a, a, a, a, a, a, a)
+
+-- | This allows users to use all the Foldable class functions on TupleN
+--   as well as all list functions like zip, map etc.
+--   I.e.  foldr acc zero (list @3 x), map f (list @5 y)
+class KnownNat n => MonoTuple (n :: Nat) a where
+  -- | This type is a bit ambiguous, so we will have to type apply it to
+  --   to a natural like this:  list @3 (3,67)
+  list :: TupleN n a -> [a]
+
+instance MonoTuple 2 a where list (x, y) = [x, y]
+instance MonoTuple 3 a where list (x, y, z) = [x, y, z]
+instance MonoTuple 4 a where list (w, x, y, z) = [w, x, y, z]
+instance MonoTuple 5 a where list (v, w, x, y, z) = [v, w, x, y, z]
+instance MonoTuple 6 a where list (u, v, w, x, y, z) = [u, v, w, x, y, z]
+instance MonoTuple 7 a where list (t, u, v, w, x, y, z) = [t, u, v, w, x, y, z]
+instance MonoTuple 8 a where list (s, t, u, v, w, x, y, z) = [s, t, u, v, w, x, y, z]
+instance MonoTuple 9 a where list (r, s, t, u, v, w, x, y, z) = [r, s, t, u, v, w, x, y, z]
+instance MonoTuple 10 a where list (q, r, s, t, u, v, w, x, y, z) = [q, r, s, t, u, v, w, x, y, z]
```

Comment:
Dead code. This is not used anywhere. Please don't tell me that it can be useful at some point in the future. If we ever need it, I am sure we can find something similar on hackage. We should not be adding code to the repo on a hypothesis that it might be useful.
```suggestion
```